docs/options api.md

```diff
@@ -0,0 +1,202 @@
+
+### DefaultOnToggle
+Like toggle but 1 (true) is the default value.
+
+### Choice
+A numeric option allowing you to define different sub options. Values are stored
+as integers but you can also do comparison methods with the class and strings so
+if you have an `option_early_sword` this can be compared with:
+```python
+if self.multiworld.sword_availability[self.player] == "early_sword":
+    do_early_sword_things()
+```
+
+or:
+```python
+from .options import SwordAvailability
+if self.multiworld.sword_availability[self.player] == SwordAvailability.option_early_sword:
+    do_early_sword_things()
+```
+
+### Range
+A numeric option allowing a variety of integers including the endpoints. Has a
+default `range_start` of 0 and default `range_end` of 1. Allows for negative
+values as well. This will always be an integer and has no methods for string
+comparisons.
+
+### SpecialRange
+Like range but also allows you to define a dictionary of special names the user
+can use to equate to a specific value. For example:
+```python
+special_range_names: {
+    "normal": 20,
+    "extreme": 99,
+}
+```
+
+will let users use the names "normal" or "extreme" in their options selections,
+but will still return those as integers to you. Useful if you want special
+handling regarding those specified values.
+
+## More Advanced Options
+### FreeText
+This is an option that allows the user to enter any possible string value. Can
+only be compared with strings, and has no validation step, so if this needs
+to be validated, you can either add a validation step to the option class or
+within the world.
+
+### TextChoice
+Like choice allows you to predetermine options and has all of the same comparison
+methods and handling. Also accepts any user defined string as a valid option,
+so will either need to be validated by adding a validation step to the option
+class or within world, if necessary. Value for this class is `Union[str, int]`
+so if you need the value at a specified point,
+`self.multiworld.my_option[self.player].current_key` will always return a string.
+
+### PlandoBosses
+An option specifically built for handling boss rando, if your game can use it.
+Is a subclass of TextChoice so supports everything it does, as long as having
+multiple validation steps to automatically support boss plando from the users.
+If using this class, you must define `bosses`, a set of valid boss names, and
+`locations`, a set of valid boss location names, and `def can_place_boss`, which
+passes a boss and location, allowing you to check if that placement is valid for
+your game. When this function is called, `bosses`, `locations`, and the passed
+strings will all be lowercase. There is also a `duplicate_bosses` attribute
+allowing you to define if a boss can be placed multiple times in your world.
+False by default, and will reject duplicate boss names from the user. For an
+example of using this class refer to `worlds.alttp.options.py`
+
+### OptionDict
```

Comment:
I'd love to see information on how to set a schema here as well. Factorio's world definition has some good examples.